### PR TITLE
fix: CE-1059 Unable to save Decision section without entering all fields - incorrect toast

### PR DESCRIPTION
**Pull request by marqueone-ps**
# Description
Unable to save Decision section without entering all fields - incorrect toast

Fixes # CE-1059

# Tasks
- [x] Add validation messages to the first 3 fields to indicate that they are required
- [x] Ensure you can save without entering anything other than the first 3 fields
- [x] Add validation when adding action input
- [x] Add validation if 
   - [x] lead agency selected
   - [x] recommend inspection 
- [x] Fix toast message

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-compliance-enforcement-629-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-compliance-enforcement-629-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-compliance-enforcement-629-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-compliance-enforcement-629-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/app/components/containers/complaints/outcomes/ceeb/ceeb-decision/decision-form.tsx [hunk 1]
```diff
@@ -222,29 +225,100 @@ export const DecisionForm: FC<props> = ({
...
+    if (!data.sector) {
+      setSectorErrorMessage("Required");
+    }
+
+    if (!data.discharge) {
+      setDischargeErrorMessage("Required");
+    }
+
+    if (!data.schedule || !data.sector || !data.discharge) {
+      _isValid = false;
```
Comment (afwilcox) [inline frontend/src/app/components/containers/complaints/outcomes/ceeb/ceeb-decision/decision-form.tsx:285]: Could cognitive complexity be reduced by 2 here by putting this into the above conditionals?

File: frontend/src/app/components/containers/complaints/outcomes/ceeb/ceeb-decision/decision-form.tsx [hunk 2]
```diff
@@ -222,29 +225,100 @@ export const DecisionForm: FC<props> = ({
...
+      if (!data.actionTakenDate) {
+        setDateActionTakenErrorMessage("Date required when action taken selected");
+        _isValid = false;
+      }
+
+      if (!data.actionTaken) {
+        setActionTakenErrorMessage("Action taken required when assigned to is selected");
+        _isValid = false;
+      }
+    }
```
Comment (afwilcox) [inline frontend/src/app/components/containers/complaints/outcomes/ceeb/ceeb-decision/decision-form.tsx:319]: If this was moved into it's own function (e.g. validateActionTaken), could cognitive complexity be reduced?

File: backend/src/v1/case_file/case_file.service.ts [hunk 3]
```diff
@@ -427,20 +427,25 @@ export class CaseFileService {
...
           caseIdentifier
+          decision { id }
         }
       }`,
       variables: { input: model },
     });
 
     const returnValue = await this.handleAPIResponse(result);
-    return returnValue?.createDecison;
+    // return returnValue?.createDecison;
```
Comment (afwilcox) [inline backend/src/v1/case_file/case_file.service.ts:437]: Removing this commented out code will remove the sonar cloud warning

**Review by afwilcox — CHANGES_REQUESTED** [review]
Two additional functionality issues that seem to be here:

- You always get the error message toast regardless of if the save is successful or not.
- If you don't fully enter the action, today's date still shows up in the read only screen.  I think there might be a fix for this in the branch CE-1059-natcom-changes but it might not be the most elegant way to resolve the problem

File: backend/src/v1/case_file/case_file.service.ts (hunk 3 above)
Comment (marqueone-ps) [reply backend/src/v1/case_file/case_file.service.ts]: this was just to help figure out why the request kept failing on me, and is cleaned up now

File: frontend/src/app/components/containers/complaints/outcomes/ceeb/ceeb-decision/decision-form.tsx (hunk 2 above)
Comment (marqueone-ps) [reply frontend/src/app/components/containers/complaints/outcomes/ceeb/ceeb-decision/decision-form.tsx]: I think this should be easier for sonar cloud though I really don't like it

**Review by afwilcox — APPROVED** [review]

**Comment by afwilcox** [comment]
Failing Cypress Test is due to a recent change in how the officer lists are populated.   Will create a bug to address.

**Pull request merged**